### ISSUE TITLE
Screen Transition Animation not working on Nested Stack Navigator

### ISSUE THREAD

**Issue by artalat** [react-navigation/react-navigation#5600]
### Current Behavior

- What code are you running and what is happening?

Snack Link: https://snack.expo.io/@artalat/react-navigation-wrapped-stack

I'm attempting to create a nested stack navigator. Moreover, I'm wrapping the nested navigator in a custom view. Code for the custom view is as follows:

```js
const NestedStack = createStackNavigator(
  {
    Home: HomeScreen,
    Details: DetailsScreen,
    Another: {
      navigationOptions: {
        header: null,
      },
      screen: createStackNavigator({
        Deep: DeepScreen,
      }),
    },
  },
  {
    initialRouteName: 'Home',
  }
);

const Comp = props => (
  <ComplexView style={{ flex: 1 }}>{props.children}</ComplexView>
);

class StackWrapper extends React.PureComponent {
  static router = NestedStack.router;

  render() {
    return (
      <Comp {...this.props}>
        <NestedStack {...this.props} />
      </Comp>
    );
  }
}

const RootStack = createStackNavigator(
  {
    Root: {
      screen: StackWrapper,
      navigationOptions: {
        header: null,
      },
    },
    Login: LoginScreen,
  },
  {
    initialRouteName: 'Root',
  }
);

const AppContainer = createAppContainer(RootStack);
```

```js
import React from 'react';
import { View } from 'react-native';

export class ComplexView extends React.Component<any> {

	render() {
		const ViewInternal2 = (props) => <View {...props} />;

		return (
			<ViewInternal2 {...this.props} />
		);
	}
}
```

When navigating to this "Details" screen, the transition animation doesn't work. The screen just abruptly changes.

When navigating to "Login" screen, the transition animation works as expected, without problems.

![react-navigation](https://user-images.githubusercontent.com/295630/52915768-bff10500-32f9-11e9-8f09-b60a9ec3505c.gif)

I have narrowed down the problem to this line: `const ViewInternal2 = (props) => <View {...props} />;`. 

Basically, in my real app, I'm wrapping a component in a HOC, and it's unavoidable because of the current architecture.

### Expected Behavior

- What do you expect should be happening?

When navigating to "Details" screen, the transition animation should work as expected.

### How to reproduce

Snack Link: https://snack.expo.io/@artalat/react-navigation-wrapped-stack

Steps:

1. Run App
2. Press "Go to Details" --> No animation
2. Press "Go to Login" --> Normal animation

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | ^3.3.0
| react-native     | ^0.57.0
| node             | 11.4.0
| npm or yarn      | yarn 1.12.3

**Comment by brentvatne** [comment]
i'm quite confused about what you're trying to accomplish with `Comp` -> `ComplexView` -- i modified it to not use that and just have some other contents in the custom stack wrapper and it works: https://snack.expo.io/@notbrent/react-navigation-wrapped-stack

**Comment by artalat** [comment]
Thanks for taking a look at it, and I understand how that must have been confusing. But that wasn't part of the problem.

I have modified my example to make it simpler. The problem is still there. Can you take another look?

https://snack.expo.io/@artalat/react-navigation-wrapped-stack

**Comment by brentvatne** [comment]
I'm still very confused about what you're trying to accomplish - what is this?

```js
import React from 'react';
import { View } from 'react-native';

export class ComplexView extends React.Component<any> {

	render() {
		const ViewInternal2 = (props) => <View {...props} />;

		return (
			<ViewInternal2 {...this.props} />
		);
	}
}
```

what is the actual use case you have?

**Comment by artalat** [comment]
This is just a simplified example of my project. This following line is actually where i use some HOC:

```js
const ViewInternal2 = (props) => <View {...props} />;
```

Here is a more realistic version:

```js
const ViewInternal2 = applyStyles(View);
```

The problem is this line. Whenever I apply any HOC, this problem occurs.

**Comment by brentvatne** [comment]
if you move `const ViewInternal2 = applyStyles(View);` out of the render function then it will work: https://snack.expo.io/@notbrent/react-navigation-wrapped-stack

**Comment by artalat** [comment]
Thanks this does seem to do the trick. 

But not sure I understand what the problem was? You've just converted into a class component. How is this different?

**Comment by brentvatne** [comment]
ah sorry that was unused in ComplexView.js, see App.js

```js
const ViewInternal2 = (props) => <View {...props} />;

class ComplexView extends React.Component<any> {
  render() {
    return <ViewInternal2 {...this.props} />;
  }
}
```

**Comment by artalat** [comment]
Ignore the comment above, i read the wrong part of the code.

The problem is fixed. 

But in my case it may not be possible to move the code outside of render function, as I need to pass the props forward in the HOC. Any other way this can be solved?

**Comment by brentvatne** [comment]
you don't have to create the react component in the render function, it's very rare that you'd need to do that and generally a bad idea. not sure why you can't just use context or props and regular component composition

**Comment by yvbeek** [comment]
In my app the stack navigation transition animation wasn't working and I was slowly losing my mind, because I couldn't figure out what was causing it.

If you are using the `IS_TESTING` environment variable, make sure you set it to `false` if you want to test the animations. Otherwise you won't see any animations at all.
